### ISSUE TITLE
Signing Support

### ISSUE THREAD

**Issue by zbindenren** [google/rpmpack#38]
Do you plan to add signing support?

**Comment by jarondl** [comment]
I did not plan to add it.

As far as I can tell, rpm signing can be done as a secondary step with `rpm --sign` [Red Hat docs](https://access.redhat.com/articles/3359321) . Is there a usecase for doing this as part of rpmpack?

**Comment by zbindenren** [comment]
We are building our rpms with this libary. The advantage would be not to shell out to rpm anymore.

**Comment by jarondl** [comment]
I guess as library usage that's ok, but I don't think this should go to the cli. Handling GPG keys is a mess, and security is a whole arena we should be careful about. Actually, maybe we can make this even simpler: Make it possible for the caller to sign the content. Then the caller is responsible for the signature part itself, and rpmpack is responsible for the format.

I think we can have an api similar to this:

```golang
func (r *RPM) SignContents(sigtag int, signer func([]byte) ([]byte, error)) error
```
Where sigtag refers to which kind of signature this is (e.g. gpg), and signer is a user supplied function that reads the data and returns the signature.

WDYS? would that make sense in your case?

**Comment by jarondl** [comment]
I have implemented this idea in #39. But I am not sure how happy I am about this solution. Please let me know if it even helps you.

**Comment by zbindenren** [comment]
@jarondl thanks, I will try it out with x/opengpg. I keep you posted.
